src/components/FullArticleCard.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getArticlesById } from "../Api";
+
+const FullArticleCard = ({ article, setArticles }) => {
+  const { article_id } = useParams();
+  //   const article = articles.find((a) => a.id === article_id);
+  useEffect(() => {
+    getArticlesById(article_id).then((data) => {
+      setArticles(data);
```

Comment:
The problem here is that instead of setting a new state for the single article, you're overwriting your state which contains all the articles.
When I view a single article and then click back to go to your homepage I get an error because as soon as I go back articles isn't an array therefore your map function isn't working.